packages/features/ee/cal-ai-phone/providers/retell-ai/RetellAIProvider.ts

```diff
@@ -0,0 +1,240 @@
+import type {
+  AIPhoneServiceProvider,
+  AIPhoneServiceProviderType,
+  AIPhoneServiceConfiguration,
+  AIPhoneServiceDeletion,
+  AIPhoneServiceDeletionResult,
+  AIPhoneServiceCallData,
+  AIPhoneServiceModel,
+  AIPhoneServiceCall,
+  AIPhoneServicePhoneNumber,
+  AIPhoneServiceAgent,
+  AIPhoneServiceUpdateModelParams,
+  AIPhoneServiceUpdateAgentParams,
+  AIPhoneServiceCreatePhoneNumberParams,
+  AIPhoneServiceImportPhoneNumberParams,
+  AIPhoneServiceUpdatePhoneNumberParams,
+  AIPhoneServiceAgentListItem,
+  AIPhoneServiceImportPhoneNumberParamsExtended,
+  AIPhoneServiceAgentWithDetails,
+} from "../../interfaces/ai-phone-service.interface";
+import { RetellAIService } from "./RetellAIService";
+import type { RetellAIRepository, RetellAgentWithDetails } from "./types";
+
+export class RetellAIProvider implements AIPhoneServiceProvider<AIPhoneServiceProviderType.RETELL_AI> {
```

Comment:
Also, maybe we should call it RetellAIPhoneServiceProvider as it makes it clear that it is the provider of phoneService and implements PhoneServiceProvider.